File: week-07/day-2/bankofsimba/src/main/java/fabian/zsofa/bankofsimba/models/BankAccount.java
package fabian.zsofa.bankofsimba.models;

public class BankAccount {
    private String name;
    private int balance;
    private String animalType;
    private boolean isKing;
    private String goodness;

    public BankAccount(String name, int balance, String animalType, boolean isKing, String goodness) {
        this.name = name;
        this.balance = balance;
        this.animalType = animalType;
        this.isKing = isKing;
        this.goodness = goodness;
    }

    public String getName() {
        return name;
    }

    public int getBalance() {
        return balance;
    }

    public String getAnimalType() {
        return animalType;
    }

    public boolean isKing() {
        return isKing;
    }

    public String getGoodness() {
        return goodness;
    }

    public void raiseBalance() {
        if (isKing) {
            this.balance += 100;
        } else {
            this.balance += 10;
        }
    }
}

File: week-11-project-phase/day-1/BankOfSimba/BankOfSimba/Models/BankAccount.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Threading.Tasks;

namespace BankOfSimba.Models
{
    public class BankAccount
    {
        private static int globalId = 1;
        public int Id { get; set; }
        public string Name { get; set; }
        public int Balance { get; set; }
        public string AnimalType { get; set; }
        public bool IsKing { get; set; }
        public bool IsGood { get; set; }

        public BankAccount()
        {
            Id = globalId++;
            IsKing = true;
            IsGood = true;
        }

        public BankAccount(string name, int balance, string animalType, bool isKing, bool isGood)
        {
            Id = globalId++;
            Name = name;
            Balance = balance;
            AnimalType = animalType;
            IsKing = isKing;
            IsGood = isGood;
        }
    }
}
